themes/dotmatrix/dotmatrix.theme

```diff
@@ -0,0 +1,895 @@
+    fi
+
+    local temperature=""
+    if _lp_temperature ; then
+        __dotmatrix_make_field "$lp_temperature" "${DOTMATRIX_COLOR_DARK[@]}"
+        temperature="${dotmatrix_field}${LP_MARK_TEMP}"
+    fi
+
+    __dotmatrix_make_sep_soft "${DOTMATRIX_COLOR_SEP_DARK[@]}"
+    # Non-bracketed variables are intentionals,
+    # so that if the variable is null, it does not count in the arguments array.
+    __dotmatrix_join_fields "$dotmatrix_sep" $battery $load $temperature
+    if [[ -n "$dotmatrix_join_fields" ]] ; then
+        side+="${blt_plug}${dotmatrix_join_fields}"
+    else
+        lp_terminal_format "${DOTMATRIX_COLOR_FAIR[@]}"
+        side+="${lp_terminal_format}"
+    fi
+
+    # jobs: detached / running / stopped
+    __dotmatrix_make_sep_plug "${DOTMATRIX_COLOR_LITE[@]}"
+    local djc_plug="${dotmatrix_sep}"
+
+    local detached_sessions=""
+    if _lp_detached_sessions ; then
+        if [[ -n "${lp_detached_sessions}" && "${lp_detached_sessions}" > 0 ]] ; then
+            __dotmatrix_make_field "${lp_detached_sessions}d" "${DOTMATRIX_COLOR_LITE[@]}"
+            detached_sessions="${dotmatrix_field}"
+        fi
+    fi
+
+    local running_jobs=""
+    local stopped_jobs=""
+    if _lp_jobcount ; then
+        if [[ -n "${lp_running_jobs}" && "${lp_running_jobs}" > 0  ]] ; then
+            __dotmatrix_make_field "${lp_running_jobs}&" "${DOTMATRIX_COLOR_LITE[@]}"
+            running_jobs="${dotmatrix_field}"
+        fi
+
+        if [[ -n "${lp_stopped_jobs}" && "${lp_stopped_jobs}" > 0  ]] ; then
+            __dotmatrix_make_field "${lp_stopped_jobs}z" "${DOTMATRIX_COLOR_LITE[@]}"
+            stopped_jobs="${dotmatrix_field}"
+        fi
+    fi
+
+    __dotmatrix_make_sep_soft "${DOTMATRIX_COLOR_SEP_LITE[@]}"
+    __dotmatrix_join_fields "$dotmatrix_sep" ${detached_sessions} ${running_jobs} ${stopped_jobs}
+    if [[ -n "$dotmatrix_join_fields" ]] ; then
+        side+="${djc_plug}${dotmatrix_join_fields}"
+    else
+        lp_terminal_format "${DOTMATRIX_COLOR_FAIR[@]}"
+        side+="${lp_terminal_format}"
+    fi
+
+    # clock
+    if _lp_time ; then
+        __dotmatrix_make_sep_plug "${DOTMATRIX_COLOR_DARK[@]}"
+        side+="${dotmatrix_sep}"
+        if _lp_analog_time ; then
+            __dotmatrix_make_field "$lp_analog_time" "${DOTMATRIX_COLOR_DARK[@]}"
+            side+="$dotmatrix_field"
+        else
+            __dotmatrix_make_field "$lp_time" "${DOTMATRIX_COLOR_DARK[@]}"
+            side+="$dotmatrix_field"
+        fi
+    fi
+
+    # Merge side in header
+    local side_as_text="$(_lp_as_text "${side}")"
+    local head_as_text="$(_lp_as_text "${header}")"
+    # FIXME within SSH, COLUMNS seems to be broken
+    local head_width=$((${COLUMNS}-${#side_as_text}-${#head_as_text}))
+    #-6)) # Why -6? I have no idea.
+    if [[ $head_width -lt 0 ]] ; then
+        head_width=0
+    fi
+
+    local spaced_side+=$(printf "%.0s " $(seq ${head_width}))
```

Comment:
Two more subshells, not good. I bet this would be faster as a for loop:
```bash
local i
for ((i=0; i < head_width; i++)); do
   spaced_side+=" "
done
```